lib/kubeclient/common.rb

```diff
@@ -296,11 +303,15 @@ def get_entity(klass, resource_name, name, namespace = nil)
       new_entity(result, klass)
     end
 
-    def delete_entity(resource_name, name, namespace = nil)
+    def delete_entity(resource_name, name, namespace = nil, options = nil)
+      params = {}
+      DELETE_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
```

Comment:
:+1:, `delete_entity(..., grace_period_seconds: 30)` is nicer interface than
 `delete_entity(..., query: {grace_period_seconds: 30})`.  

I'd like to start raising errors for unexpected options.  (Specific example I've stumbled on: hash with `:namespace` works in some methods, while others expect it positional and *silently* ignore `:namespace`, giving you little clue you called it wrong...)
But this would be a wider change, not for this PR.